browser/views.py

```diff
@@ -713,6 +713,67 @@ def delete_investigator_record(request, person_id, research_id):
     return HttpResponseRedirect(reverse('person', args=(person_id,)))
 
 
+@staff_member_required
+def edit_affiliation(request, person_id, affiliation_id):
+    person = get_object_or_404(Person, pk=person_id)
+    affiliation = get_object_or_404(Affiliation, pk=affiliation_id)
+    template = "browser/affiliation_edit.html"
+
+    form = EditAffiliationForm(initial={'institution': affiliation.institution,
+                                        'position': affiliation.position,
+                                        'institution_id': affiliation.institution.id if affiliation.institution else None})
+    context = {
+        'form': form,
+        'person': person
+    }
+
+    if request.method == 'POST':
+        form = EditAffiliationForm(request.POST, instance=person)
+        success, institution = _handle_institution(form, request)
+        if success:
+            Affiliation.objects.filter(id=affiliation_id).update(position=form.cleaned_data.get('position'),
+                                                     institution=institution)
+            return HttpResponseRedirect(reverse('person', args=(person_id,)))
+        else:
+            messages.add_message(request, messages.ERROR,
+                                 'The above institution does not exist in the database. '
+                                 'Please check the checkbox below to create a new institution')
+            context.update({'form': form})
+            return render(request, template, context)
+
+    return render(request, template, context)
+
+
+@staff_member_required
+def delete_affiliation_record(request, person_id, affiliation_id):
+    if request.method == 'POST':
+        Affiliation.objects.filter(id=affiliation_id).delete()
+
+    return HttpResponseRedirect(reverse('person', args=(person_id,)))
+
+
+def _handle_institution(form, request):
+    """
+    Checks if an institution is to be created or an existing institution is to be retrieved
+    """
+    institution = None
+    if form.is_valid():
```

Comment:
what if the form is not valid?

Reply:
If the form is not valid, it goes back to the parent class and displays the error on the form. I verified it by changing one of the form's required field to True and tried submitting it.

Reply:
ya, since you're not returning anything, success isn't true. But that's pretty bad practice as it is very unexpected.